src/status_im/contexts/wallet/data_store.cljs

```diff
@@ -187,3 +188,80 @@
      :removed-account-addresses   removed-account-addresses
      :updated-keypairs-by-id      updated-keypairs-by-id
      :updated-accounts-by-address updated-accounts-by-address}))
+
+(defn- rename-keys-to-kebab-case
+  [m]
+  (set/rename-keys m (zipmap (keys m) (map transforms/->kebab-case-keyword (keys m)))))
+
+(defn rpc->suggested-routes
```

Comment:
@ilmotta I already tried rewriting the `cske/transform-keys` function to do less work, although I've got better results I wasn't sure it was the best approach. For example, that function is adding metadata with the previous values, if we skip those operations, the transformations are faster, although the main problem is using a Clojure walker.

We could try exploring a performant solution, but I honestly think it's going to be expensive to catch all cases

Reply:
Got it @ulisesmac 👍🏼 I benchmarked the recursive implementation from @BalogunofAfrica in this PR and it performs significantly better than `cske/transform-keys`, although of course it's of limited usage. But `cske/transform-keys` I think shouldn't be used.

In fact I think we shouldn't be doing recursive key transformations even with our own code because the hand-rolled code with raw maps (not using `rename-keys` or `update-keys`) to transform keys is considerably faster as far as benchmarked. I thought we had an agreement about this, but then found we're using in the wallet context. That's okay, I was just surprised :)

Also it's not like we need to rename keys all the time from status-go, usually we're careful with that nowadays because it's a breaking change. So we might be optimizing too much for ergonomics in a more sensitive part of the stack.

I'm copying @cammellos and @flexsurfer just to see if they are okay with this. If not we should probably discuss on Discord and not here because there are already many comments.

Thanks for the insights as usual @ulisesmac

Reply:
sorry for the late reply, I would be leaning towards avoiding dynamic transformation of keys, we had to remove them in at least 2/3 instances on performance grounds, so I would be wary of using them, albeit of course they provide a great dev experience